flyway-core/src/main/java/org/flywaydb/core/internal/util/scanner/classpath/ClassPathScanner.java

```diff
@@ -192,38 +193,43 @@ private String toClassName(String resourceName) {
                 if ("file".equals(url.getProtocol())
                         && url.getPath().endsWith(".jar")
                         && !url.getPath().matches(".*" + Pattern.quote("/jre/lib/") + ".*")) {
-                    // All non-system jars on disk
-                    JarFile jarFile;
                     try {
```

Comment:
GitHub's diff did a pretty poor job of recognizing the actual change here.  I simply wrapped the existing code in a try-catch.